File: src/Webs/daotao/src/domain/course_class.ts

export interface CourseClass {
    index: number;
    weekStart: number;
    weekEnd: number;
    status: number;
    courseClassCode: string;
    courseClassName: string;
    studentIds: string[];
    courseClassType: number;
    subjectCode: string;
    sessionLengths: number[];
    totalSession: number;
    semesterCode: string | null;
    numberStudents: number;
    numberStudentsExpected: number;
    stage: number;
    teacherName: string | null;
    teacherCode: string | null;
    parentCourseClassCode: string;
    id: string;
    createdAt: string;
    updatedAt: string | null;
}

export interface SlotTimeRegister {
    BuildingCode: string;
    RoomCode: string;
    DayOfWeek: number;
    Slot: string[]; 
}

export interface CourseClassRegister {
    CourseClassCode: string;
    CourseClassName: string;
    CourseClassType: number;
    SubjectCode: string;
    SubjectName: string;
    NumberOfCredits: number;
    TeacherCode: string;
    TeacherName: string;
    SemesterCode: string;
    Students: string[] | null;
    Stage: number;
    SlotTimes: SlotTimeRegister[];
}

File: src/Services/RegisterService/RegisterStudy.Domain/CourseClass.cs
namespace RegisterStudy.Domain;

public class CourseClass
{
    public string CourseClassCode { get; set; }
    public string CourseClassName { get; set; }
    public int CourseClassType { get; set; }
    public string SubjectCode { get; set; }
    public string SubjectName { get; set; }
    public int NumberOfCredits { get; set; }
    public string TeacherCode { get; set; }
    public string TeacherName { get; set; }
    public DateTime StartDate { get; set; }
    
    public DateTime EndDate { get; set; }
    public string SemesterCode { get; set; }
    public int NumberStudentsExpected { get; set; }
    public List<string> Students { get; set; } = [];
    public int Stage { get; set; }
    
    public string ParentCourseClassCode { get; set; }
    public int WeekStart { get; set; } = 0;
    public int WeekEnd { get; set; } = 0;
    
    public List<SlotTime> SlotTimes { get; set; } = [];

}

public record SlotTime(
    string BuildingCode,
    string RoomCode,
    int DayOfWeek,
    List<string> Slot,
    int WeekStart,
    int WeekEnd
);